internal/transport/http2_client.go

```diff
@@ -1190,10 +1198,11 @@ func (t *http2Client) handleData(f *http2.DataFrame) {
 		// guarantee f.Data() is consumed before the arrival of next frame.
 		// Can this copy be eliminated?
 		if len(f.Data()) > 0 {
-			buffer := t.bufferPool.get()
-			buffer.Reset()
-			buffer.Write(f.Data())
-			s.write(recvMsg{buffer: buffer})
+			pool := s.ct.bufferPool
+			if pool == nil {
+				pool = mem.DefaultBufferPool()
+			}
```

Comment:
Similar here, we should get rid of the `nil` check.

And we can directly use the buffer pool from the transport here, instead of getting to the transport from the stream and then getting the buffer pool.

Reply:
I added a comment on this, it will never be nil except for tests.